### PR TITLE
Register reflective classes from bean archive index by providing the index

### PR DESCRIPTION
**Pull request by mswiderski**
register reflective classes from bean archive index by providing the index as part of the reflective class build item.

this is another change needed for https://github.com/quarkusio/quarkus/pull/2675 to make this work on native image as well. It simply does not take into account the bean archive items for generated classes. In addition int core-deployment we can't use arc build items so I added the index to reflective class build item instead. 

Wondering if better would be to use first the combined index and as fallback the one from build item (https://github.com/quarkusio/quarkus/compare/master...mswiderski:missing-reflective-classes?expand=1#diff-eb505e9693864a419f6cf2426bf540e2R117)

Obviously I'd like to have your opinion @gsmet @mkouba

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
It should definitely check both indexes if they are available.

**Comment by mswiderski** [comment]
@stuartwdouglas is this approach valid or you would suggest other way to check both indexes?

**Comment by mswiderski** [comment]
is there any feedback on this? It is prerequisite for making kogito quarkus extension to work in native image, see this PR https://github.com/quarkusio/quarkus/pull/2752

**Comment by cescoffier** [comment]
Looks good to me, waiting for @stuartwdouglas comments/approval.

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**
